skrifa/src/outline/glyf/mod.rs

```diff
@@ -827,25 +937,267 @@ impl<'a> Scaler<'a> {
+        let points_end = points_start + point_count + PHANTOM_POINT_COUNT;
+        let point_range = points_start..points_end;
+        // Points and flags are accumulated as we load the outline.
+        let points = self
+            .memory
+            .points
+            .get_mut(point_range.clone())
+            .ok_or(InsufficientMemory)?;
+        let flags = self
+            .memory
+            .flags
+            .get_mut(point_range)
+            .ok_or(InsufficientMemory)?;
+        glyph.read_points_fast(&mut points[..point_count], &mut flags[..point_count])?;
+        // Compute the range for our contour end point buffer and slice it.
+        let contours_start = self.contour_count;
+        let contour_end_pts = glyph.end_pts_of_contours();
+        let contour_count = contour_end_pts.len();
+        let contours_end = contours_start + contour_count;
+        let contours = self
+            .memory
+            .contours
+            .get_mut(contours_start..contours_end)
+            .ok_or(InsufficientMemory)?;
+        // Read the contour end points.
+        for (end_pt, contour) in contour_end_pts.iter().zip(contours.iter_mut()) {
+            *contour = end_pt.get();
+        }
+        // Adjust the running point/contour total counts
+        self.point_count += point_count;
+        self.contour_count += contour_count;
+        // Append phantom points to the outline.
+        for (i, phantom) in self.phantom.iter().enumerate() {
+            points[phantom_start + i] = *phantom;
+            flags[phantom_start + i] = Default::default();
+        }
+        // Acquire deltas
+        if self.outlines.gvar.is_some() && !self.coords.is_empty() {
+            let gvar = self.outlines.gvar.as_ref().unwrap();
+            let glyph = deltas::SimpleGlyph {
+                points: &mut points[..],
+                flags: &mut flags[..],
+                contours,
+            };
+            let deltas = self
+                .memory
+                .deltas
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            let iup_buffer = self
+                .memory
+                .iup_buffer
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            if deltas::simple_glyph(
+                gvar,
+                glyph_id,
+                self.coords,
+                self.outlines.has_var_lsb,
+                glyph,
+                iup_buffer,
+                deltas,
+            )
+            .is_ok()
+            {
+                for (point, delta) in points.iter_mut().zip(deltas) {
+                    *point += *delta;
+                }
+            }
+        }
+        // Apply scaling
+        if self.is_scaled {
+            let scale = self.scale.to_f32();
+            for point in points.iter_mut() {
+                *point = point.map(|c| c * scale);
+            }
+        }
```

Comment:
I didn't really look into how HB applies scale because it wasn't necessary for icons. This probably needs to be slotted in at the correct place to maintain compatibility.

Reply:
Deferred to https://github.com/googlefonts/fontc/issues/784